src/test/test_pidfile.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash -e
+
+#
+# test pidfile here 
+#
+
+# Includes
+source ../qa/workunits/ceph-helpers.sh
+
+function run() {
+    local dir=$1
+    shift
+
+    export CEPH_MON="127.0.0.1:17108"
+    export CEPH_ARGS
+    CEPH_ARGS+="--fsid=$(uuidgen) --auth-supported=none "
+    CEPH_ARGS+="--mon-host=$CEPH_MON "
+
+    local funcs=${@:-$(set | sed -n -e 's/^\(TEST_[0-9a-z_]*\) .*/\1/p')}
+    for func in $funcs ; do
+        $func $dir || return 1
+    done
+}
+
+
+function TEST_normal() {
+    local dir=$1
+    setup $dir 
+    run_mon $dir a 
+    run_osd $dir 0
+    teardown $dir 
+}
+
+function TEST_without_pidfile() {
+    local dir=$1
+    setup $dir
+    local RUNID=`uuidgen`
+    run_mon $dir a --pid-file= --daemonize=$RUNID || { teardown_unexist_pidfile $dir; return 1; } 
+    run_osd $dir 0 --pid-file= --daemonize=$RUNID || { teardown_unexist_pidfile $dir; return 1; }
+    teardown_unexist_pidfile $dir $RUNID || return 1
+}
+
+function teardown_unexist_pidfile() {
+    local dir=$1
+    shift
+    local RUNID=$1
+    shift
+    local delays=('0' '1' '3' '5' '10' '10' '20' '60')
+    local pids=$(ps aux|awk "/cep[h].*$RUNID.*/ {print \$2}")
+    for i in $pids ; do
+        local kill_complete="0"
+        local count=0
+        while [ $kill_complete -eq "0" ] ; do  
+            if kill -9 $i 2> /dev/null ; then
+                kill_complete="0"
+                count=$((count+1))
+                sleep ${delays[$count]}
+            else
+                kill_complete="1"
+            fi
+       done
+    done
+    if [ $(stat -f -c '%T' .) == "btrfs" ]; then
+         __teardown_btrfs $dir
+    fi
+    rm -fr $dir
+    if [ $kill_complete -eq "0" ] ;then
```

Comment:
so only the last `kill_complete` is recored. what if we failed to kill the first pid after tried 9 times, and killed the last one successfully ?